waspc/data/Generator/templates/server/src/middleware.ts

```diff
@@ -0,0 +1,41 @@
+{{={= =}=}}
+import express from 'express'
+import cookieParser from 'cookie-parser'
+import logger from 'morgan'
+import cors from 'cors'
+import helmet from 'helmet'
+
+import config from './config.js'
+
+{=# globalMiddlewareConfigFnDefined =}
+{=& globalMiddlewareConfigFnImportStatement =}
+{=/ globalMiddlewareConfigFnDefined =}
+{=^ globalMiddlewareConfigFnDefined =}
+const {=& globalMiddlewareConfigFnImportAlias =} = (m: MiddlewareConfig) => m
+{=/ globalMiddlewareConfigFnDefined =}
+
+export type MiddlewareConfig = Map<string, express.RequestHandler>
+
+export type MiddlewareConfigFn = (middleware: MiddlewareConfig) => MiddlewareConfig
+
+const _defaultMiddleware: MiddlewareConfig = new Map([
+  ['helmet', helmet()],
+  ['cors', cors({ origin: config.allowedCORSOrigins })],
+  ['logger', logger('dev')],
+  ['express.json', express.json()],
+  ['express.urlencoded', express.urlencoded({ extended: false })],
+  ['cookieParser', cookieParser()]
+])
+
+const defaultMiddleware = {=& globalMiddlewareConfigFnImportAlias =}(_defaultMiddleware)
+
+export function getDefaultMiddleware(): MiddlewareConfig {
```

Comment:
This is really a global middleware, right? So we should probably call it that way.
Also, it is not really default, is it? Because it can get modified by that globalMiddlewareConfigFnImportAlias function, so then it is not default any more?

Reply:
File should probably also be renamed. Actually what I guess is confusing that in this file concepts of middleware in general + global middleware are mixed?

When dealing with API routes you handle modification of default middleware right there, in place, while for global middleware do you do it in here in this file -> I would maybe also do that rather directly in place, or create a separate file/module for it.

Reply:
Renamed file and variable names. Created one entrypoint for mods.